buildifier/buildifier.go

```diff
@@ -319,41 +348,22 @@ func processFile(filename string, data []byte, inputType, lint string, warningsL
 	case "check":
 		// check mode: print names of files that need formatting.
 		if !bytes.Equal(data, ndata) {
-			// Print:
-			//	name # list of what changed
-			reformat := ""
-			if !bytes.Equal(data, beforeRewrite) {
-				reformat = " reformat"
-			}
-			log := ""
-			if len(info.Log) > 0 && *showlog {
-				sort.Strings(info.Log)
-				var uniq []string
-				last := ""
-				for _, s := range info.Log {
-					if s != last {
-						last = s
-						uniq = append(uniq, s)
-					}
-				}
-				log = " " + strings.Join(uniq, " ")
-			}
-			fmt.Printf("%s #%s %s%s\n", filename, reformat, &info, log)
+			fileDiagnostics.MarkNotFormatted()
+			fileDiagnostics.SetRewrites(info.Stats())
 			exitCode = 4
 		}
-		return
 
 	case "diff":
 		// diff mode: run diff on old and new.
 		if bytes.Equal(data, ndata) {
-			return
+			break
```

Comment:
instead of all these break statements (which require to to follow down to see what happens next

make these all
return fileDiagnostics, exitCode


and in the case where it was

exitCode = 3
return

use 
return fileDiagnostics, 3

Reply:
Done.